source/code/plugins/in_sudo_tail.rb

```diff
@@ -0,0 +1,135 @@
+require 'yajl'
+require 'fluent/input'
+require 'fluent/event'
+require 'fluent/config/error'
+require 'fluent/parser'
+require 'open3'
+
+module Fluent
+  class SudoTail < Input
+    Plugin.register_input('sudo_tail', self)
+
+    def initialize
+      super
+      @command = nil
+    end
+
+    attr_accessor :command
+
+    #The command (program) to execute.
+    config_param :path, :string
+
+    #The format used to map the program output to the incoming event.
+    config_param :format, :string, default: 'none'
+
+    #Tag of the event.
+    config_param :tag, :string, default: nil
+
+    #Fluentd will record the position it last read into this file.
+    config_param :pos_file, :string, default: nil
+
+    #The interval time between periodic program runs.
+    config_param :run_interval, :time, default: nil
+
+    #Start to read the log from the head of file.  
+    config_param :read_from_head, :bool, default: false
+
+    def configure(conf)
+      super
+      unless @path
+        raise ConfigError, "'path' parameter is not set to a 'tail' source."
+      end
+      
+      unless @pos_file
+        raise ConfigError, "'pos_file' is required to keep track of file"
+      end 
+
+      unless @tag 
+        raise ConfigError, "'tag' is required on sudo tail"
+      end
+
+      unless @run_interval
+        raise ConfigError, "'run_interval' is required for periodic tailing"      
+      end
+ 
+      @parser = Plugin.new_parser(conf['format'])
+      @parser.configure(conf)
+
+      ruby = '/opt/microsoft/omsagent/ruby/bin/ruby '
+      tailscript = '/opt/microsoft/omsagent/bin/tailfilereader.rb '
+      @command = "sudo " << ruby << tailscript << @path <<  " -p #{@pos_file}" 
```

Comment:
This is going to run the default ruby version on the system which may differ wildly from the one shipped with oms.